aws/resource_aws_ecs_service.go

```diff
@@ -321,6 +328,15 @@ func resourceAwsEcsServiceCreate(d *schema.ResourceData, meta interface{}) error
 		input.LaunchType = aws.String(v.(string))
 	}
 
+	if v, ok := d.GetOk("scheduling_strategy"); ok {
```

Comment:
Should `deploymentConfiguration` be taken into account too?

> If deploymentConfiguration is specified, the maximum percent parameter must be 100. The default value for a daemon service for maximumPercent is 100%. The default value for a daemon service for minimumHealthyPercent is 0% for the AWS CLI, the AWS SDKs, and the APIs, and 50% for the AWS Management Console.

https://docs.aws.amazon.com/AmazonECS/latest/developerguide/ecs_services.html#service_scheduler_daemon

Reply:
I was thinking about that too, but I'm not sure what the general handlings of this are in other places. I'm obviously relying on AWS doing the validation and returning the error for this. But if the best practice here is to do the validation on the terraform side, I can add it. Let me know.

Reply:
The `scheduling_strategy` attribute is always going to exist since it has a `Default` -- you can use `input.SchedulingStrategy = d.Get("scheduling_strategy").(string)` instead of working with `d.GetOk()` or add it to the `CreateEcsService` struct directly with the same `d.Get()`